src/resources/parser/glb-container-resource.js

```diff
@@ -190,6 +191,41 @@ class GlbContainerResource {
         return GlbContainerResource.createSceneHierarchy(sceneClones, 'Entity');
     }
 
+    // get material variants
+    getMaterialVariants() {
+        return this.data.variants;
+    }
+
+    // apply material variant to entity
+    applyMaterialVariant(name, entity) {
+        const variant = this.data.variants[name];
+        if (!variant)
+            return;
+        const renders = entity.findComponents("render");
+        for (let i = 0; i < renders.length; i++) {
+            const renderComponent = renders[i];
+            renderComponent.meshInstances.forEach((instance) => {
+                const meshVariants = this.data.meshVariants[instance.mesh.id];
+                if (meshVariants) {
+                    instance.material = this.data.materials[meshVariants[variant]];
```

Comment:
I'd just add line like this after this one, to make sure this holds true:
`Debug.Assert(instance.material);`

Reply:
also, could this function call applyMaterialVariantInstances to avoid duplication of the same code?